plugins/net_plugin/net_plugin.cpp

```diff
@@ -3025,7 +3025,8 @@ namespace eosio {
          return true;
       }
       if (my_impl->sync_master->syncing_from_peer()) {
-         peer_wlog(this, "syncing, dropping trx");
+         peer_dlog(this, "syncing, dropping trx");
+         pending_message_buffer.advance_read_ptr( message_length );
```

Comment:
Should we have a test for this, now that wlog is changed to dlog? (would the wlog have helped identify the failure?)

Reply:
Peers should not send you trx when you are syncing. However, they may not realize you are syncing especially when you first connect. Our integration tests will still log this as they run in full debug logging mode.

Reply:
I meant a test that shows the buffer does advance when this happens.

Reply:
Unfortunately the net_plugin is not currently setup for unittests. Would be nice to refactor net_plugin to allow for these types of tests. As is there is no way to force net_plugin to send trxs when syncing.